### ISSUE TITLE
gradle 8.8 build failed with missing java libs in /app/libs

### ISSUE THREAD

**Issue by FtenBosch** [ktsr42/RsyncServerApp#11]
**TLDR:**
"Build is missing 3 java libs:
/RsyncServerApp/app/libs/yajsync-app-0.9.1.jar
/RsyncServerApp/app/libs/yajsync-core-0.9.1.jar
/RsyncServerApp/app/libs/libintf-0.9.1.jar
Didn't find them in APK or in the original yajsync java build or anywhere online. At a loss where to get them."

**Detailed version:**

Hi! I tried to build the app (because I changed the default Port and ModuleName to static defaults) using the following configuration:

```
# gradle --version
------------------------------------------------------------
Gradle 8.8
------------------------------------------------------------
Build time:   2024-05-31 21:46:56 UTC
Revision:     4bd1b3d3fc3f31db5a26eecb416a165b8cc36082

Kotlin:       1.9.22
Groovy:       3.0.21
Ant:          Apache Ant(TM) version 1.10.13 compiled on January 4 2023
JVM:          21.0.2 (Oracle Corporation 21.0.2+13-58)
OS:           Linux 6.5.0-41-generic amd64

```

Using a current Android studio:

> Android Studio Koala | 2024.1.1
> Build #AI-241.15989.150.2411.11948838, built on June 10, 2024
> Runtime version: 17.0.10+0-17.0.10b1087.21-11609105 amd64
> VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
> Linux 6.5.0-41-generic
> GC: G1 Young Generation, G1 Old Generation
> Memory: 2048M
> Cores: 8
> Registry:
>   ide.experimental.ui=true
> Current Desktop: ubuntu:GNOME

When executing gradle build, I get the following error (redacted for personal dir paths):

```
# gradle build --warning-mode all                                                      

> Configure project :app                    
Declaring client module dependencies has been deprecated. This is scheduled to be removed in Gradle 9.0. Please use component metadata rules instead. Consult the upgrading guide for further information: https://docs.gradle.org/8.8/userguide/upgrading_version_8.html#declaring_client_module_dependencies
        at build_6y1qab9qlrv65qde54x7gs3jy.run(<REDACTED_PERSONAL_DIRECTORY_PATH>/RsyncServerApp/app/build.gradle:1)
        (Run with --stacktrace to get the full stack trace of this deprecation warning.)

> Task :app:processDebugResources UP-TO-DATE
Mutating the dependencies of configuration ':app:debugCompileClasspath' after it has been resolved or consumed. This behavior has been deprecated. This will fail with an error in Gradle 9.0. After a Configuration has been resolved, consumed as a variant, or used for generating published metadata, it should not be modified. Consult the upgrading guide for further information: https://docs.gradle.org/8.8/userguide/upgrading_version_8.html#mutate_configuration_after_locking

> Task :app:compileDebugJavaWithJavac FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileDebugJavaWithJavac'.
> Could not resolve all files for configuration ':app:debugCompileClasspath'.
   > Failed to transform libintf-0.9.1.jar to match attributes {artifactType=android-classes-jar, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for IdentityTransform: <REDACTED_PERSONAL_DIRECTORY_PATH>/RsyncServerApp/app/libs/libintf-0.9.1.jar.
         > File/directory does not exist: <REDACTED_PERSONAL_DIRECTORY_PATH>/RsyncServerApp/app/libs/libintf-0.9.1.jar
   > Failed to transform yajsync-app-0.9.1.jar to match attributes {artifactType=android-classes-jar, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for IdentityTransform: <REDACTED_PERSONAL_DIRECTORY_PATH>/RsyncServerApp/app/libs/yajsync-app-0.9.1.jar.
         > File/directory does not exist: <REDACTED_PERSONAL_DIRECTORY_PATH>/RsyncServerApp/app/libs/yajsync-app-0.9.1.jar
   > Failed to transform yajsync-core-0.9.1.jar to match attributes {artifactType=android-classes-jar, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for IdentityTransform: <REDACTED_PERSONAL_DIRECTORY_PATH>/RsyncServerApp/app/libs/yajsync-core-0.9.1.jar.
         > File/directory does not exist: <REDACTED_PERSONAL_DIRECTORY_PATH>/RsyncServerApp/app/libs/yajsync-core-0.9.1.jar

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 1s
13 actionable tasks: 1 executed, 12 up-to-date
```

So in essence, it's missing 3 java libs:
/RsyncServerApp/app/libs/yajsync-app-0.9.1.jar
/RsyncServerApp/app/libs/yajsync-core-0.9.1.jar
/RsyncServerApp/app/libs/libintf-0.9.1.jar

I also successfully built the original yajsync Server using maven:
https://github.com/perlundq/yajsync

But it did not yield these libraries either. All I got were those:
```
# find  . -name "*.jar"
./yajsync-core/target/yajsync-core-0.9.0-SNAPSHOT.jar
./yajsync-app/target/yajsync-app-0.9.0-SNAPSHOT-full.jar
./yajsync-app/target/yajsync-app-0.9.0-SNAPSHOT.jar
```
DuckDuck & Google didn't help. I am at a loss where to find these libs. Would be nice if you could point me into the right direction. Thanks a lot in advance!

**Comment by FtenBosch** [comment]
Nevermind... I just found my mistake. I've build the wrong yajsync server. 

I should've built yours:
https://github.com/ktsr42/yajsync

... and then take your ...
https://github.com/ktsr42/yajsync/blob/master/push_android.sh

... as a template to create symlinks into /app/libs
